BoltsTest/src/bolts/TaskTest.java

```diff
@@ -268,7 +270,251 @@ public void testWhenAllNoTasks() {
+            assertFalse(task.isCancelled());
+            assertEquals(task.getResult(), firstToCompleteError);
+            assertTrue(task.getResult().isCompleted());
+            assertFalse(task.getResult().isCancelled());
+            assertTrue(task.getResult().isFaulted());
+            return null;
+          }
+        });
+      }
+    });
+  }
 
+  public void testWhenAnyResultFirstCancelled() {
+    runTaskTest(new Callable<Task<?>>() {
+      @Override
+      public Task<?> call() throws Exception {
+        final ArrayList<Task<Void>> tasks = new ArrayList<Task<Void>>();
+        final Task<Void> firstToCompleteCancelled = Task.callInBackground(new Callable<Void>() {
+          @Override
+          public Void call() throws Exception {
+            Thread.sleep(50);
+            throw new CancellationException();
+          }
+        });
+         
+        tasks.addAll(launchTasksWithRandomCompletions(10));
+        tasks.add(firstToCompleteCancelled);
+        return Task.whenAnyResult(tasks).continueWith(new Continuation<Task<Void>, Object>() {
+          @Override
+          public Object then(Task<Task<Void>> task) throws Exception {
+            assertTrue(task.isCompleted());
+            assertFalse(task.isFaulted());
+            assertFalse(task.isCancelled());
+            assertEquals(task.getResult(), firstToCompleteCancelled);
+            assertTrue(task.getResult().isCompleted());
+            assertTrue(task.getResult().isCancelled());
+            assertFalse(task.getResult().isFaulted());
+            return null;
+          }
+        });
+      }
+    });
+  }
+    
+  public void testWhenAnyFirstCancelled() {
+    runTaskTest(new Callable<Task<?>>() {
+      @Override
+      public Task<?> call() throws Exception {
+        final ArrayList<Task<Void>> tasks = new ArrayList<Task<Void>>();
+        final Task<Void> firstToCompleteCancelled = Task.callInBackground(new Callable<Void>() {
+          @Override
+          public Void call() throws Exception {
+            Thread.sleep(50);
+            throw new CancellationException();
+          }
+        });
+        tasks.addAll(launchTasksWithRandomCompletions(10));
+        tasks.add(firstToCompleteCancelled);
+        return Task.whenAny(tasks).continueWith(new Continuation<Task<?>, Object>() {
+          @Override
+          public Object then(Task<Task<?>> task) throws Exception {
+            assertTrue(task.isCompleted());
+            assertFalse(task.isFaulted());
+            assertFalse(task.isCancelled());
+            assertEquals(task.getResult(), firstToCompleteCancelled);
+            assertTrue(task.getResult().isCompleted());
+            assertTrue(task.getResult().isCancelled());
+            assertFalse(task.getResult().isFaulted());
+            return null;
+          }
+        });
+      }
+    });
+  }
+
+  private Collection<Task<Void>> launchTasksWithRandomCompletions(int numberOfTasksToLaunch ) {
```

Comment:
Let's add documentation here to describe what this does.

nit: removing spacing after `numberOfTasksToLaunch`